test/unit/range/views/translate_test.cpp

```diff
@@ -105,76 +77,33 @@ TYPED_TEST(nucleotide, view_translate_single)
-    seqan3::aa27_vector cmp1{"TYVR"_aa27};
-    seqan3::aa27_vector cmp1_rev{"YVRT"_aa27};
-    seqan3::aa27_vector cmp1b{"RTYV"_aa27};
-    seqan3::aa27_vector cmp1c{"VRT"_aa27};
-    seqan3::aa27_vector cmp1c_rev{"RTY"_aa27};
-    seqan3::aa27_vector cmp2{"CMHA"_aa27};
-    seqan3::aa27_vector cmp3{"AHMC"_aa27};
+    auto const vec = std::string_view{"ACGTACGTACGTA"} | seqan3::views::char_to<TypeParam>;
 
     // default parameter translation_frames
-    auto const v1 = vec | seqan3::views::translate_single;
-    // == [T,Y,V,R]
-    EXPECT_EQ(v1.size(), cmp1.size());
-    EXPECT_RANGE_EQ(v1, cmp1);
+    EXPECT_RANGE_EQ("TYVR"_aa27, vec | seqan3::views::translate_single);
 
     // default parameter translation_frames
-    auto const v2 = vec | seqan3::views::translate_single();
-    // == [T,Y,V,R]
-    EXPECT_EQ(v2.size(), cmp1.size());
-    EXPECT_RANGE_EQ(v2, cmp1);
+    EXPECT_RANGE_EQ("TYVR"_aa27, vec | seqan3::views::translate_single());
 
     // single frame translation
-    auto const v3 = vec | seqan3::views::translate_single(seqan3::translation_frames::FWD_FRAME_0);
-    // == [T,Y,V,R]
-    EXPECT_EQ(v3.size(), cmp1.size());
-    EXPECT_RANGE_EQ(v3, cmp1);
-    auto const v3_rev = vec | seqan3::views::translate_single(seqan3::translation_frames::REV_FRAME_0);
-    // == [Y,V,R,T]
-    EXPECT_EQ(v3_rev.size(), cmp1_rev.size());
-    EXPECT_RANGE_EQ(v3_rev, cmp1_rev);
-    auto const v3b = vec | seqan3::views::translate_single(seqan3::translation_frames::FWD_FRAME_1);
-    // == [R,T,Y,V]
-    EXPECT_EQ(v3b.size(), cmp1b.size());
-    EXPECT_RANGE_EQ(v3b, cmp1b);
-    auto const v3b_rev = vec | seqan3::views::translate_single(seqan3::translation_frames::REV_FRAME_1);
-    // == [T,Y,V,R]
-    EXPECT_EQ(v3b_rev.size(), cmp1.size());
-    EXPECT_RANGE_EQ(v3b_rev, cmp1);
-    auto const v3c = vec | seqan3::views::translate_single(seqan3::translation_frames::FWD_FRAME_2);
-    // == [V,R,T]
-    EXPECT_EQ(v3c.size(), cmp1c.size());
-    EXPECT_RANGE_EQ(v3c, cmp1c);
-    auto const v3c_rev = vec | seqan3::views::translate_single(seqan3::translation_frames::REV_FRAME_2);
-    // == [R,T,Y]
-    EXPECT_EQ(v3c_rev.size(), cmp1c_rev.size());
-    EXPECT_RANGE_EQ(v3c_rev, cmp1c_rev);
+    EXPECT_RANGE_EQ("TYVR"_aa27, vec | seqan3::views::translate_single(seqan3::translation_frames::FWD_FRAME_0));
+    EXPECT_RANGE_EQ("YVRT"_aa27, vec | seqan3::views::translate_single(seqan3::translation_frames::REV_FRAME_0));
+    EXPECT_RANGE_EQ("RTYV"_aa27, vec | seqan3::views::translate_single(seqan3::translation_frames::FWD_FRAME_1));
+    EXPECT_RANGE_EQ("TYVR"_aa27, vec | seqan3::views::translate_single(seqan3::translation_frames::REV_FRAME_1));
+    EXPECT_RANGE_EQ("VRT"_aa27, vec | seqan3::views::translate_single(seqan3::translation_frames::FWD_FRAME_2));
+    EXPECT_RANGE_EQ("RTY"_aa27, vec | seqan3::views::translate_single(seqan3::translation_frames::REV_FRAME_2));
```

Comment:
You are getting coverage drops because you do not test `size()` anymore, both for const and non-const version of `seqan3::views::translate_single`

Reply:
https://app.codecov.io/gh/seqan/seqan3/compare/2443/changes


There's two missed throw in `operator[]` and in `size()`, even though we test it...

Reply:
Interesting, I guess I will add the test template for ranges. That should test all of this.